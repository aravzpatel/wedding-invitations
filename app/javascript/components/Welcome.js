import React from "react"
import PropTypes from "prop-types"
class Welcome extends React.Component {
  constructor(props){
    super(props);

    this.listNames = this.listNames.bind(this)
  }

  listNames(){
    var guests = this.props.guests;
    var names = [];
    names = guests.map((guest) => guest.first_name);
    var guestList;

    if(names.length > 1){
      return guestList = names.splice(0, names.length-1).join(', ') + ' and '+ names[names.length - 1]
    } else {
      return guestList = names[0]
    }
  }

  render () {
    return (
      <div className = 'welcome_greeting'>
        <h1>
          {this.listNames()}
        </h1>
      </div>
    );
  }
}

Welcome.propTypes = {
  guests: PropTypes.array
};

export default Welcome
