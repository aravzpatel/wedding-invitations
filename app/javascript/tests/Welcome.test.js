import React from "react";
import Welcome from "../components/Welcome";
import renderer from "react-test-renderer";

test("Welcome component creates a single string of 1 user", () => {
  const guests = [{first_name: "Jack"}];
  const component = renderer.create(
    <Welcome guests = {guests}/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Welcome component combines two users into one string", () => {
  const guests = [{first_name: "Jack"}, {first_name: "Jill"}];
  const component = renderer.create(
    <Welcome guests = {guests}/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});