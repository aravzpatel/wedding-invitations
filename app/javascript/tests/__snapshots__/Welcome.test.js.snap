// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component combines two users into one string 1`] = `
<div
  className="welcome-greeting"
>
  <div>
    Jack and Jill
  </div>
</div>
`;

exports[`Welcome component creates a single string of 1 user 1`] = `
<div
  className="welcome-greeting"
>
  <div>
    Jack
  </div>
</div>
`;
